File: docs/guides/swift/Sources/searchFuturePublishedBooks.swift
import Foundation
#if os(Linux) // For linux interop
    import FoundationNetworking
#endif

import Core
import Search

func searchFuturePublishedBooks() async throws {
    do {
        let client = try SearchClient(appID: "ALGOLIA_APPLICATION_ID", apiKey: "ALGOLIA_API_KEY")

        let dateTimestamp = Int(Date().timeIntervalSince1970)
        let searchParams = SearchSearchParams.searchSearchParamsObject(
            SearchSearchParamsObject(query: "<YOUR_SEARCH_QUERY>", filters: "date_timestamp > \(dateTimestamp)")
        )

        let response: SearchResponse<Hit> = try await client.searchSingleIndex(
            indexName: "<YOUR_INDEX_NAME>",
            searchParams: searchParams
        )
        print(response)
    } catch {
        print(error)
    }
}

File: docs/guides/kotlin/src/main/kotlin/com/algolia/snippets/searchFuturePublishedBooks.kt
import java.time.Instant

import com.algolia.client.api.SearchClient
import com.algolia.client.configuration.*
import com.algolia.client.transport.*
import com.algolia.client.extensions.*
import com.algolia.client.model.search.*

suspend fun searchFuturePublishedBooks() {
  val client = SearchClient(appId = "ALGOLIA_APPLICATION_ID", apiKey = "ALGOLIA_API_KEY")

  val dateTimestamp = Instant.now().epochSecond
  val searchParams = SearchParamsObject(
    query = "<YOUR_SEARCH_QUERY>",
    filters = "date_timestamp > $dateTimestamp",
  )

  client.searchSingleIndex(indexName = "<YOUR_INDEX_NAME>", searchParams = searchParams)
}
